src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1561,9 +1561,19 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> {
                 await broadcaster.broadcastLoadEventsForAll(this.expressionMap.mainAlias.target, entities);
         }
 
+        const virtualColumns = this.expressionMap.selects
+            .filter(select => select.virtual)
+            .map(select => select.selection.replace(this.alias + ".", ""));
+
+        // remove virtual columns from the entity
+        entities = entities.map(entity => {
+            virtualColumns.forEach(virtualColumn => delete entity[virtualColumn]);
+            return entity;
+        });
```

Comment:
I think this logic should be somewhere inside raw result to entity transformer, because here you remove virtuals from top-level entities. Add test like this:

```ts
createQueryBuilder("post")
.select(["post.title", "post.description")
.addSelect(["category.title"])
.leftJoin("post.categories", "category")
.getMany();
```

you should except following data:

```json
[{
   title: "..",
   description: "..",
   categories: [{
      title: "..",
   }]
}]
```

Something like that